src/App.tsx

```diff
@@ -1,16 +1,143 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { Error } from './types/Error';
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  changeTodo,
+} from './api/todos';
+
+import { TodoList } from './components/TodoList/TodoList';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { TodoError } from './components/TodoError/TodoError';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [userTodos, setUserTodos] = useState<Todo[]>([]);
+  const [hasTodos, setHasTodos] = useState(false);
+  const [hasError, setHasError] = useState<Error>({ status: false });
+  const [isAdding, setIsAdding] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [changingIds, setChangingIds] = useState<number[]>([0]);
+
+  const handleSetError = useCallback((error: Error) => {
+    setHasError(error);
+    setTimeout(() => {
+      setHasError({ status: false });
+    }, 3000);
+  }, []);
+
+  const handleLoadTodos = useCallback(async () => {
+    try {
+      setHasError({ status: false });
+
+      const todosFromServer = await getTodos((user as User).id);
+
+      setUserTodos(todosFromServer);
+    } catch {
+      handleSetError({ status: true });
+    }
+  }, []);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    handleLoadTodos();
+  }, []);
+
+  useEffect(() => {
+    if (userTodos.length !== 0) {
+      setHasTodos(true);
+    } else {
+      setHasTodos(false);
+    }
+  }, [userTodos]);
```

Comment:
Why do you need this state? You can just compare todos length to 0 and save this result to constant